src/smtsolvers/CoreSMTSolver.cc

```diff
@@ -1897,3 +1897,28 @@ void CoreSMTSolver::fillBooleanVars(ModelBuilder &modelBuilder) {
         modelBuilder.addVarValue(atom, val);
     }
 }
+
+vec<CRef> CoreSMTSolver::getUnsatCoreClauses() const {
+    assert(resolutionProof);
+    vec<CRef> unsatCore;
+    auto const & derivations = resolutionProof->getProof();
+    std::vector<CRef> stack {CRef_Undef};
```

Comment:
I think that
```
std::vector<CRef> stack;
stack.push_back(CRef_Undef);
```
is actually more efficient, because this could lead to a better estimate of the reserved size. `{CRef_Undef}` can lead to reserving just one element which just leads to unavoidable reallocation.
It would be even better if there is a good estimate of the max. size, e.g.
```
stack.reserve(derivations.size());
```

Reply:
I'll skip the reserve here.